### PR TITLE
Add endpoint to get all users with a specific rank

### PR DESCRIPTION
**Pull request by draeeeeeeeeeeeem**
I haven't tested this yet as I haven't gotten the API setup on a server, although I'm pretty confident it'll work.

### CODE REVIEW

**Comment by draeeeeeeeeeeeem** [comment]
Regarding [this suggestion](https://github.com/WarzoneMC/Website/issues/23), can be utilized to add a staff page.

**Comment by jellz** [comment]
why are you returning 401

File: routes/minecraft/ranks.js [hunk 1]
```diff
@@ -12,6 +12,20 @@ module.exports = function (app) {
             res.json(ranks);
         })
     });
+    
+    app.get('/mc/players/:rank', (req, res) => {
+        MinecraftRank.find({ name: req.params.rank.toLowerCase() }).limit(1).exec((err, ranks) => {
+            var rank = ranks[0];
+            
+            if (!rank) {
+                res.status(401).json({ notFound: true });
```
Comment (jellz) [inline routes/minecraft/ranks.js]: Why are you returning 401?

Comment (draeeeeeeeeeeeem) [reply routes/minecraft/ranks.js]: im copying the other endpoints, the one just below it uses 401 for notfound

Comment (MattBSG) [reply routes/minecraft/ranks.js]: HTTP 401 is very different from HTTP 404.

HTTP 401: Unauthorized - you do not have the proper permission to view a resource
HTTP 404: Not Found - the resource is unavailable or does not exist

I’ll open a PR later today to bring the rest up to code and uniform. You should use 404 here

Comment (jellz) [reply routes/minecraft/ranks.js]: Exactly what I thought! Thank you. Someone needs to teach the API developers a lesson HTTP, haha.

**Comment by draeeeeeeeeeeeem** [comment]
Updated: switched 401 to 404 in accordance with [Matt's PR](https://github.com/WarzoneMC/API/pull/8). Only reason I originally used 401 was to keep style with the rest of the code, which was also using 401.

**Comment by draeeeeeeeeeeeem** [comment]
Tested code on local version and can confirm it works.

**Comment by lukechatton** [comment]
Endpoint should more clearly show that it's a query by rank.

**Comment by draeeeeeeeeeeeem** [comment]
Yeah I realized that, on my local version I've updated

**Pull request merged**